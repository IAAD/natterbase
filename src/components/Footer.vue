<template>
    <div class="footer_container">
        <footer class="footer">
            <div class="container">
                <div class="row">
                    <div class="col-sm-3">
                        <h4 class="title">PAYSMOSMO</h4>
                        <p class="footer_subtitle">Help give out inventory Loans to cooperative members with ease</p>

                    </div>
                    <div class="col-sm-2">
                        <h4 class="title2">Contact</h4>

                        <p><a href="#"> Registration</a></p>
                        <p><a href="#"> About Us</a></p>
                        <p><a href="#"> Contact Us</a></p>

                    </div>
                    <div class="col-sm-2">
                        <h4 class="title2">Contact</h4>

                        <p><a href="#"> Registration</a></p>
                        <p><a href="#"> About Us</a></p>
                        <p><a href="#"> Contact Us</a></p>

                    </div>
                    <div class="col-sm-3">
                        <h4 class="title2">Stay Connected</h4>
                        <font-awesome-layers class="fa-2x" full-width>
                            <font-awesome-icon :icon="{ prefix: 'fab', iconName: 'facebook' }" pull="left" />
                            <font-awesome-layers-text  transform="down-2 shrink-8" value="7" />
                        </font-awesome-layers>
                        <font-awesome-layers class="fa-2x" full-width>
                            <font-awesome-icon :icon="{ prefix: 'fab', iconName: 'twitter' }"  pull="left"  />
                            <font-awesome-layers-text  transform="down-2 shrink-9" value="9" />
                        </font-awesome-layers>
                        <font-awesome-layers class="fa-2x" full-width>
                            <font-awesome-icon :icon="{ prefix: 'fab', iconName: 'instagram' }"  pull="left"  />
                            <font-awesome-layers-text transform="down-2 shrink-8" value="5" />
                        </font-awesome-layers>

                    </div>
                    <div class="col-sm-2">
                        <h4 class="title2">Back to top ^</h4>
                    </div>
                </div>

            </div>


        </footer>
        <div class="text-center footer_down">
            <p class="footer_at"> <a href=""> Â© 2018 PAYSMOSMO </a> </p>
        </div>

    </div>
</template>

<script>
    export default {
        name: "Footer"
    }
</script>

<style scoped>

</style>