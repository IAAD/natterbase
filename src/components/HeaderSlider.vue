<template>
    <b-container fluid>
        <b-row>
            <div class="col-lg-3 col-md-3 col-sm-12">
                <b-list-group >
                    <b-list-group-item class="category_card bg-success"> CATEGORIES </b-list-group-item>
                    <div v-bind:key="category.id" v-for="category in categories">
                    <b-list-group-item >{{category.name}} <i class="fa-pull-right"> > </i></b-list-group-item>
                    </div>
                </b-list-group>
            </div>

            <div class="col-lg-9 col-md-9 col-sm-12">
                    <div>
                        <b-carousel
                                id="carousel1"
                                style="text-shadow: 1px 1px 2px #333;"
                                indicators
                                background="#ababab"
                                :interval="4000"
                                img-width="1024"
                                img-height="536"
                                v-model="slide"
                                @sliding-start="onSlideStart"
                                @sliding-end="onSlideEnd"
                        >


                            <!-- Slides with custom text -->
                            <b-carousel-slide img-src="https://picsum.photos/1024/536/?image=370">
                                <div class="slide_text">
                                <h1 class="slide_main_text">NEW COLLECTION</h1>
                                    <p>It is a long established fact that a reader will be distracted by the readable <br>
                                    content of a page when looking at its layout
                                    </p>

                                    <button class="btn btn-outline-warning fa-pull-left"> SHOP NOW</button>
                                </div>
                            </b-carousel-slide>

                            <!-- Slides with custom text -->
                            <b-carousel-slide img-src="https://picsum.photos/1024/536/?image=370">
                                <div class="slide_text">
                                    <h1 class="slide_main_text">NEW COLLECTION</h1>
                                    <p>It is a long established fact that a reader will be distracted by the readable <br>
                                        content of a page when looking at its layout
                                    </p>

                                    <button class="btn btn-outline-warning fa-pull-left"> SHOP NOW</button>
                                </div>
                            </b-carousel-slide>

                            <!-- Slides with custom text -->
                            <b-carousel-slide img-src="https://picsum.photos/1024/536/?image=370">
                                <div class="slide_text">
                                    <h1 class="slide_main_text">NEW COLLECTION</h1>
                                    <p>It is a long established fact that a reader will be distracted by the readable <br>
                                        content of a page when looking at its layout
                                    </p>

                                    <button class="btn btn-outline-warning fa-pull-left"> SHOP NOW</button>
                                </div>
                            </b-carousel-slide>

                            <!-- Slides with custom text -->
                            <b-carousel-slide img-src="https://picsum.photos/1024/536/?image=370">
                                <div class="slide_text">
                                    <h1 class="slide_main_text">NEW COLLECTION</h1>
                                    <p>It is a long established fact that a reader will be distracted by the readable <br>
                                        content of a page when looking at its layout
                                    </p>

                                    <button class="btn btn-outline-warning fa-pull-left"> SHOP NOW</button>
                                </div>
                            </b-carousel-slide>

                            <!-- Slides with custom text -->
                            <b-carousel-slide img-src="https://picsum.photos/1024/536/?image=370">
                                <div class="slide_text">
                                    <h1 class="slide_main_text">NEW COLLECTION</h1>
                                    <p>It is a long established fact that a reader will be distracted by the readable <br>
                                        content of a page when looking at its layout
                                    </p>

                                    <button class="btn btn-outline-warning fa-pull-left"> SHOP NOW</button>
                                </div>
                            </b-carousel-slide>






                        </b-carousel>

                    </div>
            </div>
        </b-row>
    </b-container>
</template>

<script>
    export default {
        name: "HeaderSlider",
        data() {
            return {
                slide: 0,
                sliding: null
            }
        },
        props:['categories'],
        methods: {
            onSlideStart() {
                this.sliding = true
            },
            onSlideEnd() {
                this.sliding = false
            }
        }
    }
</script>

<style scoped>
    .list-group-item{
        text-align: left !important;
        font-size: 14px;
        border-radius:0px !important;
    }
    .category_card{
        color: white;
    }
    .carousel-indicators li{
        width: 10px !important;
        height: 10px!important;
        border-radius: 100% !important;
    }
    .carousel-indicators li:active{
       background: #573322;
    }
    .carousel-item:before {
        content: '';
        position: absolute;
        top: 0;
        right: 0;
        bottom: 0;
        left: 0;
        height: 536px;
        width: 100%;
        background-image: linear-gradient(to bottom right,#242526,#2a2525);
        opacity: .8;
    }

    .slide_main_text{
        text-align: left;

    }

    .slide_text{
        margin-bottom: 25%;
        margin-left: -16%;
    }

    .slide_text p{
        text-align: left;
    }
    @media screen and (max-width: 550px){
        .slide_text{
            margin-bottom: 15%;
            margin-left: -5%;
        }
        .slide_text p{
            font-size:8px;
            margin-bottom: 0px;
        }

        .btn-outline-warning{
            padding: 0px;
        }
        .slide_text h1{
            font-size:10px;
        }


    }

</style>