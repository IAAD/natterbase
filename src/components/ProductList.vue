<template>
    <div class="top_explore">
        <div >
            <b-card-group >
                <b-card
                        img-src="https://picsum.photos/600/300/?image=370"
                        img-alt="Image"
                        img-top
                        tag="article"
                        style="max-width: 20rem;"
                        class="mb-2 card_style"
                        v-for="item in items" v-bind:key="item.id"
                >
                    <p class="card_text">{{item.name}}</p>
                    <star-rating :rating=item.rating :star-size="20" :read-only="true" ></star-rating>
                    <p class="price_text"><span class="last_price">N{{item.old_price}}</span> <span class="fa-pull-right">N{{item.price}}</span></p>
                </b-card>

            </b-card-group>

        </div>

    </div>
</template>

<script>
    export default {
        name: "ProductList",
        props:['items']
    }
</script>

<style scoped>

</style>