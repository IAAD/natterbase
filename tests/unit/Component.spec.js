import { mount } from '@vue/test-utils'
import HeaderSlider from '@/components/HeaderSlider.vue'
import Footer from '@/components/Footer.vue'
import ProductList from '@/components/ProductList.vue'



describe('HeaderSlider', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(HeaderSlider)
    expect(wrapper.isVueInstance()).toBeTruthy()

  })
})

describe('HeaderSlider', () => {
  test('renders Correctly', () => {
    const wrapper = mount(HeaderSlider)
    expect(wrapper.element).toMatchSnapshot()

  })
})

describe('Footer', () => {
  test('Footer is a Vue instance', () => {
    const wrapper = mount(Footer)
    expect(wrapper.isVueInstance()).toBeTruthy()

  })
})

describe('Footer', () => {
  test('Footer renders Correctly', () => {
    const wrapper = mount(Footer)
    expect(wrapper.element).toMatchSnapshot()

  })
})

describe('ProductList', () => {
  test('ProductList is a Vue instance', () => {
    const wrapper = mount(ProductList)
    expect(wrapper.isVueInstance()).toBeTruthy()

  })
})

describe('ProductList', () => {
  test('ProductList renders Correctly', () => {
    const wrapper = mount(ProductList)
    expect(wrapper.element).toMatchSnapshot()

  })
})