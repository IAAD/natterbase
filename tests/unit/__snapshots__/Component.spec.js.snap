// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Footer renders Correctly 1`] = `
<div
  class="footer_container"
>
  <footer
    class="footer"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-3"
        >
          <h4
            class="title"
          >
            PAYSMOSMO
          </h4>
           
          <p
            class="footer_subtitle"
          >
            Help give out inventory Loans to cooperative members with ease
          </p>
        </div>
         
        <div
          class="col-sm-2"
        >
          <h4
            class="title2"
          >
            Contact
          </h4>
           
          <p>
            <a
              href="#"
            >
               Registration
            </a>
          </p>
           
          <p>
            <a
              href="#"
            >
               About Us
            </a>
          </p>
           
          <p>
            <a
              href="#"
            >
               Contact Us
            </a>
          </p>
        </div>
         
        <div
          class="col-sm-2"
        >
          <h4
            class="title2"
          >
            Contact
          </h4>
           
          <p>
            <a
              href="#"
            >
               Registration
            </a>
          </p>
           
          <p>
            <a
              href="#"
            >
               About Us
            </a>
          </p>
           
          <p>
            <a
              href="#"
            >
               Contact Us
            </a>
          </p>
        </div>
         
        <div
          class="col-sm-3"
        >
          <h4
            class="title2"
          >
            Stay Connected
          </h4>
           
          <font-awesome-layers
            class="fa-2x"
            full-width=""
          >
            <font-awesome-icon
              icon="[object Object]"
              pull="left"
            />
             
            <font-awesome-layers-text
              transform="down-2 shrink-8"
              value="7"
            />
          </font-awesome-layers>
           
          <font-awesome-layers
            class="fa-2x"
            full-width=""
          >
            <font-awesome-icon
              icon="[object Object]"
              pull="left"
            />
             
            <font-awesome-layers-text
              transform="down-2 shrink-9"
              value="9"
            />
          </font-awesome-layers>
           
          <font-awesome-layers
            class="fa-2x"
            full-width=""
          >
            <font-awesome-icon
              icon="[object Object]"
              pull="left"
            />
             
            <font-awesome-layers-text
              transform="down-2 shrink-8"
              value="5"
            />
          </font-awesome-layers>
        </div>
         
        <div
          class="col-sm-2"
        >
          <h4
            class="title2"
          >
            Back to top ^
          </h4>
        </div>
      </div>
    </div>
  </footer>
   
  <div
    class="text-center footer_down"
  >
    <p
      class="footer_at"
    >
      <a
        href=""
      >
         Â© 2018 PAYSMOSMO 
      </a>
    </p>
  </div>
</div>
`;

exports[`HeaderSlider renders Correctly 1`] = `
<b-container
  fluid=""
>
  <b-row>
    <div
      class="col-lg-3 col-md-3 col-sm-12"
    >
      <b-list-group>
        <b-list-group-item
          class="category_card bg-success"
        >
           CATEGORIES 
        </b-list-group-item>
         
      </b-list-group>
    </div>
     
    <div
      class="col-lg-9 col-md-9 col-sm-12"
    >
      <div>
        <b-carousel
          background="#ababab"
          id="carousel1"
          img-height="638"
          img-width="1024"
          indicators=""
          interval="4000"
          style="text-shadow: 1px 1px 2px #333;"
        >
          <b-carousel-slide
            img-src="https://picsum.photos/1024/638/?image=370"
          >
            <div
              class="slide_text"
            >
              <h1
                class="slide_main_text"
              >
                NEW COLLECTION
              </h1>
               
              <p>
                It is a long established fact that a reader will be distracted by the readable 
                <br />
                
                                content of a page when looking at its layout
                                
              </p>
               
              <button
                class="btn btn-outline-warning fa-pull-left"
              >
                 SHOP NOW
              </button>
            </div>
          </b-carousel-slide>
           
          <b-carousel-slide
            img-src="https://picsum.photos/1024/638/?image=370"
          >
            <div
              class="slide_text"
            >
              <h1
                class="slide_main_text"
              >
                NEW COLLECTION
              </h1>
               
              <p>
                It is a long established fact that a reader will be distracted by the readable 
                <br />
                
                                    content of a page when looking at its layout
                                
              </p>
               
              <button
                class="btn btn-outline-warning fa-pull-left"
              >
                 SHOP NOW
              </button>
            </div>
          </b-carousel-slide>
           
          <b-carousel-slide
            img-src="https://picsum.photos/1024/638/?image=370"
          >
            <div
              class="slide_text"
            >
              <h1
                class="slide_main_text"
              >
                NEW COLLECTION
              </h1>
               
              <p>
                It is a long established fact that a reader will be distracted by the readable 
                <br />
                
                                    content of a page when looking at its layout
                                
              </p>
               
              <button
                class="btn btn-outline-warning fa-pull-left"
              >
                 SHOP NOW
              </button>
            </div>
          </b-carousel-slide>
           
          <b-carousel-slide
            img-src="https://picsum.photos/1024/638/?image=370"
          >
            <div
              class="slide_text"
            >
              <h1
                class="slide_main_text"
              >
                NEW COLLECTION
              </h1>
               
              <p>
                It is a long established fact that a reader will be distracted by the readable 
                <br />
                
                                    content of a page when looking at its layout
                                
              </p>
               
              <button
                class="btn btn-outline-warning fa-pull-left"
              >
                 SHOP NOW
              </button>
            </div>
          </b-carousel-slide>
           
          <b-carousel-slide
            img-src="https://picsum.photos/1024/638/?image=370"
          >
            <div
              class="slide_text"
            >
              <h1
                class="slide_main_text"
              >
                NEW COLLECTION
              </h1>
               
              <p>
                It is a long established fact that a reader will be distracted by the readable 
                <br />
                
                                    content of a page when looking at its layout
                                
              </p>
               
              <button
                class="btn btn-outline-warning fa-pull-left"
              >
                 SHOP NOW
              </button>
            </div>
          </b-carousel-slide>
        </b-carousel>
      </div>
    </div>
  </b-row>
</b-container>
`;

exports[`ProductList ProductList renders Correctly 1`] = `
<div
  class="top_explore"
>
  <div>
    <b-card-group />
  </div>
</div>
`;
